kratos/processes/compute_nodal_normal_divergence_process.cpp

```diff
@@ -293,7 +230,37 @@ void ComputeNodalNormalDivergenceProcess<ComputeNodalDivergenceProcessSettings::
 /***********************************************************************************/
 /***********************************************************************************/
 
+template <bool THistorical>
+void ComputeNodalNormalDivergenceProcess<THistorical>::ComputeDivergence(
+    const Geometry<Node<3>>& rGeometry,
+    const Matrix& rDN_DX,
+    const Variable<array_1d<double,3>>& rVariable,
+    double& rDivergence
+    )
+{
+    if (mNonHistoricalOriginVariable) {
+        for(std::size_t i_node=0; i_node<rGeometry.size(); ++i_node) {
+            const auto& vector_field = rGeometry[i_node].GetValue(rVariable);
+
+            const double norm = norm_2(vector_field) + std::numeric_limits<double>::epsilon();
```

Comment:
Please avoid this. I know your concerns, but the main aim of this process was to calculate curvature so the norm should be non-zero.
```suggestion
            const double norm = norm_2(vector_field);
            #ifdef KRATOS_DEBUG
            KRATOS_WARNING_IF("NodalNormalDivergenceProcess", norm < 1.0e-12) << "WARNING: unexpected zero norm" <<std::endl;
            #endif
```

Reply:
I would remove the `WARNING`, since `KRATOS_WARNING` already prints one

Reply:
it's avoided and a warning is added